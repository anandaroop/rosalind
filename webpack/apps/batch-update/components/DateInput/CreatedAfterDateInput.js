import React from 'react'
import DateInput from './DateInput'

const CreatedAfterDateInput = (props) => {
  return (
    <DateInput
      onSelectDate={props.onSelectDate}
      date={props.createdAfterDate}
      placeholder='Created after date'
    />
  )
}

export { CreatedAfterDateInput }
