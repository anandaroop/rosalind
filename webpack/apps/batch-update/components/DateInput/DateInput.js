import React from 'react'
import chrono from 'chrono-node'
import '../Styles/Input.css'

export default class DateInput extends React.Component {
  constructor (props) {
    super(props)
    this.handleChange = this.handleChange.bind(this)
    this.state = {
      date: null
    }
  }

  handleChange (event) {
    const parsed = parseDate(event.target.value)
    this.props.selectDate(parsed)
    this.setState({
      date: parsed
    })
  }

  render () {
    return (
      <div>
        <input
          type='text'
          onChange={this.handleChange}
          />
        <div className='parsed'>{this.state.date}</div>
      </div>
    )
  }
}

function parseDate (raw) {
  const result = chrono.parse(raw)
  if (result.length > 0) {
    return result[0].start.date().toString()
  } else {
    return null
  }
}
