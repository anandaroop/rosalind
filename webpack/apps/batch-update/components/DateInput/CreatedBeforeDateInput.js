import React from 'react'
import DateInput from './DateInput'

const CreatedBeforeDateInput = (props) => {
  return (
    <DateInput
      onSelectDate={props.onSelectDate}
      date={props.createdBeforeDate}
      placeholder='Created before date'
    />
  )
}

export default CreatedBeforeDateInput
