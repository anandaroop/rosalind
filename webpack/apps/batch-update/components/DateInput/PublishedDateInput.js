import React from 'react'
import DateInput from './DateInput'

const PublishedDateInput = (props) => {
  return (
    <div>
      <DateInput
        selectDate={props.onSelectPublishedDate}
        />
    </div>
  )
}

export default PublishedDateInput
