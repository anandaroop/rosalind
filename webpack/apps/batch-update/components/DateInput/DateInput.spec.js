import React from 'react'
import renderer from 'react-test-renderer'
import DateInput from './DateInput'

it('renders correctly', () => {
  const rendered = renderer.create(<DateInput />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
