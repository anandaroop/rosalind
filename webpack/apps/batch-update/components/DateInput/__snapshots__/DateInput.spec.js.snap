exports[`test renders correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text" />
  <div
    className="parsed" />
</div>
`;
